src/compiler/transformers/es2015.ts

```diff
@@ -2758,10 +2758,6 @@ export function transformES2015(context: TransformationContext): (x: SourceFile
             if (assignments) {
                 updated = setTextRange(factory.createExpressionStatement(factory.inlineExpressions(assignments)), node);
             }
-            else {
-                // none of declarations has initializer - the entire variable statement can be deleted
-                updated = undefined;

```

Comment:
alternatively, emptied labeled statements could be removed by [this function](https://github.com/microsoft/TypeScript/blob/451388cfd5b0d086c5f69b9ff5675b650feef623/src/compiler/transformers/es2015.ts#L2942-L2950). This would perhaps make sense for `labeledStatementDeclarationListInLoopNoCrash1` and `labeledStatementDeclarationListInLoopNoCrash2`. When it comes to the other ones... they are so malformed that I'm not so sure. I think that TS responsibility isn't necessarily to minify the output and preserving nodes structure in cases like this is just safer